### PR TITLE
implement put_u8 for BytesMut

### PR DESCRIPTION
**Pull request by seanmonstar**
Benching before and after shows that this implementation is about twice as fast as going through `put_slice`.

### CODE REVIEW

**Comment by carllerche** [comment]
You could probably make it even faster by only having one `is_inline()` check :) You can roughly see what I mean here: https://github.com/carllerche/bytes/pull/82

Do you want to try that out?

**Comment by seanmonstar** [comment]
Yes, reducing the `is_inline` to one call makes it a little over 2x faster than this proposed implementation, for a total ~4x faster...

File: src/bytes.rs [hunk 1]
```diff
@@ -1463,6 +1475,21 @@ impl Inner {
         }
     }
 
+    /// Insert a byte into the next slot and advance the len by 1.
+    #[inline]
+    unsafe fn put_u8(&mut self, n: u8) {
```
Comment (carllerche) [inline src/bytes.rs]: Why is this function unsafe? While the implementation requires `unsafe`, I don't think that unsafety leaks out?

**Comment by seanmonstar** [comment]
Laziness :). Fixed.

**Pull request merged**